modules/core/src/main/java/org/apache/ignite/internal/client/thin/ReliableChannel.java

```diff
@@ -678,7 +675,7 @@ synchronized void initChannelHolders() {
 
         int idx = curChIdx;
 
-        if (idx != -1)
+        if (idx != -1 && holders != null)
```

Comment:
Looks like, that in case when `idx != -1` then `holders` never `null`. Could you please describe the scenario?